nemo/collections/asr/models/msdd_models.py

```diff
@@ -1408,8 +1417,114 @@ def run_overlap_aware_eval(
                 out_rttm_dir=self.out_rttm_dir,
             )
             output = score_labels(
-                self.AUDIO_RTTM_MAP, all_reference, all_hypothesis, collar=collar, ignore_overlap=ignore_overlap,
+                rttm_map, all_reference, all_hypothesis, collar=collar, ignore_overlap=ignore_overlap,
             )
             outputs.append(output)
         logging.info(f"  \n")
         return outputs
+
+    @classmethod
+    def from_pretrained(
+        cls,
+        diar_model_name: str = 'diar_msdd_telephonic',
+        vad_model_name: str = 'vad_multilingual_marblenet',
+        speaker_model_name: str = 'titanet_large',
+        domain: Optional[str] = None,
+        device: Union[torch.device, str] = "auto",
+    ):
+        """
+        Instantiate a `NeuralDiarizer` to run Speaker Diarization.
+
+        Args:
+            diar_model_name (str): Path/Name of the neural diarization model to load.
+            vad_model_name (str): Path/Name of the voice activity detection (VAD) model to load.
+            speaker_model_name (str): Path/Name of the speaker model to load.
+            domain (str): Specific domain of thea udio, supports "telephonic" and "meeting" (default is general).
+            device (torch.device): Device we are running on. "auto selects CUDA if available ('cpu', 'cuda').
+
+        Returns:
+            `NeuralDiarizer`
+        """
+        cfg = NeuralInferenceConfig.from_domain(domain, device)
+        cfg.diarizer.msdd_model.model_path = diar_model_name
+        cfg.diarizer.vad.model_path = diar_model_name
+        cfg.diarizer.speaker_embeddings.model_path = diar_model_name
+        return cls(cfg)
+
+    def __call__(
+        self,
+        inference_input: Union[str, List[str]],
+        max_speakers: Optional[int] = None,
+        num_speakers: Optional[int] = None,
```

Comment:
It made me question if we should support multiple files at all. The issue with this is it just makes the API a bit confusing; I think it just makes more logical sense to run diarization multiple times IMO from a user perspective. Can be swayed due to the optimization though.

Reply:
Agree. Lets make it to support only one file inference and if users would like to do on multiple files, then they can just run the example script.

Reply:
+1 Agreed on one-file inference this time.

Reply:
How costly is it to run the 3-4 model N times vs calling them exactly N/BS time ? I would go for multiple file support but it's your domain so I'll defer to you.

Reply:
Good question, but let's just support a single file for now, would be easy to add support in a separate iteration.

Reply:
I get so many comments about long-form audio diarization. Current diarization implementation maxes out batch size for embedding extraction (for multiple files) then clustering is done for single files since 1h audio can be 3G GPU mem and 2h can be 10G GPU mem. So I think processing one-by-one is not a bad decision for lay-people oriented API.

Reply:
Removed multi-file support